holopy/scattering/interface.py

```diff
@@ -355,3 +351,47 @@ def scattered_field_to_hologram(scat, ref):
     holo = (np.abs(total_field.sel(vector=['x', 'y']))**2).sum(dim=vector)
     return holo
 
+
+def _choose_mie_vs_multisphere(spheres):
+    center_or_radius_not_set = [
+        getattr(s, k) is None
+        for s in spheres.scatterers for k in ['center', 'r']]
+    if len(spheres.scatterers) == 1:
+        theory = Mie()
+    elif any(center_or_radius_not_set):
+        msg = ("Sphere centers and radii must be set for scattering " +
+               "calculations with more than one sphere.")
+        raise InvalidScatterer(spheres, msg)
+    elif any([not np.isscalar(sphere.r) for sphere in spheres.scatterers]):
+        warn("HoloPy's multisphere theory can't handle coated spheres." +
+             "Using Mie theory.")
+        theory = Mie()
+    else:
+        # We choose Multisphere if the spheres are close enough, else
+        # Mie superposition. We choose the theory that is most accurate.
+        # What is close enough? From Jerome's paper (
+        # the effects from multiple scattering are on the order of
+        #       error_mie ~ Q_ext *(ka)^2 / kR
+        # For large spheres, Q_ext -> 2, so we can write this as
+        #       error_mie ~ kR * (a / R)^2 << 1
+        # The Multisphere theory uses spherical harmonic translation
+        # theorems to expand out each particle's scattered field and
+        # self-consistently solve for multiple-sphere scattering.
+        # For computational reasons Multisphere does not go past a fixed order
+        # (=70), but the actual number of terms needed scales as kR.
+        # So for large R, the error in Multisphere scales as
+        #       error_multisphere = kR / 100
+        # where the factor of 100 is the value of kR at which
+        # multisphere stops increasing the expansion order.
+        # The error for multisphere is smaller than the Mie
+        # superposition error when
+        #       (a / R)^2 / 100 < 1
```

Comment:
I think this might be backwards? This statement has multisphere error being smaller than Mie for very large R but I think that's the opposite of what we want.

Reply:
Sorry, I'm bad at math. I clarified the confused derivation I had there; it should be fixed in e5cf24e5